source/grid/grid_generator.cc

```diff
@@ -28,11 +30,1180 @@
+                  const int    ix = std::round(std::abs(node(0) - H(0)) / dx);
+                  const int    iy = std::round(std::abs(node(1)) / dy);
+
+                  const double alpha_y = bias_alpha(1 - 1.0 * iy / n_cells_y);
+                  const double alpha_x =
+                    bias_alpha(1 - ((double)ix) / n_cells_x_2);
+                  // define on upper/lower horizontal far field side at y = +/-
+                  // height, i.e. face GK or IL incline factor to move points G
+                  // and H to the right
+                  const Point<2> p1(
+                    J(0) - (1 - incline_factor) * length_b2 * (alpha_x),
+                    ((cell.material_id() == 3) ? (height) : (-height)));
+                  // define points on HJ but use tail_y as y-coordinate, in case
+                  // last airfoil point has y =/= 0
+                  const Point<2> p2(J(0) - alpha_x * length_b2, tail_y);
+                  node = p1 * (1 - alpha_y) + p2 * alpha_y;
+                }
+              else
+                {
+                  Assert(false, ExcIndexRange(cell.material_id(), 1, 6));
+                }
+            }
+      }
+
+
+      /*
+       * This function returns a bias factor 'alpha' which is used to make the
+       * mesh more tight in close distance of the airfoil.
+       *
+       * @param alpha
+       * @return alpha
+       */
+      double
+      bias_alpha(double alpha) const
+      {
+        return std::tanh(bias_factor * alpha) / std::tanh(bias_factor);
+      }
+    };
+
+
+
+    template <>
+    void create_triangulation(Triangulation<1, 1> &, const AdditionalData &)
+    {
+      Assert(false, ExcMessage("Airfoils only exist for 2D and 3D!"));
+    }
+
+
+
+    template <>
+    void create_triangulation(Triangulation<1, 1> &,
+                              std::vector<GridTools::PeriodicFacePair<
+                                typename Triangulation<1, 1>::cell_iterator>> &,
+                              const AdditionalData &)
+    {
+      Assert(false, ExcMessage("Airfoils only exist for 2D and 3D!"));
+    }
+
+
+
+    template <>
+    void create_triangulation(Triangulation<2, 2> & tria,
+                              const AdditionalData &additional_data)
+    {
+      MeshGenerator mesh_generator(additional_data);
+
+      if (auto parallel_tria =
+            dynamic_cast<dealii::parallel::distributed::Triangulation<2, 2> *>(
+              &tria))
+        mesh_generator.create_triangulation(*parallel_tria, nullptr);
+      else if (auto parallel_tria = dynamic_cast<
+                 dealii::parallel::fullydistributed::Triangulation<2, 2> *>(
+                 &tria))
+        mesh_generator.create_triangulation(*parallel_tria, nullptr);
+      else
+        mesh_generator.create_triangulation(tria, nullptr);
+      // Assert(false, ExcMessage("Mesh has been created successfully!"));
+      std::cout << "Mesh has been created successfully!" << std::endl;
+    }
```

Comment:
I changed this function as you suggested, but there is a compiling error which says no matching function to call, since 

no known conversion for argument 2 from ‘std::nullptr_t’ to ‘std::vector<dealii::GridTools::PeriodicFacePair<dealii::TriaIterator<dealii::CellAccessor<2, 2> > >, std::allocator<dealii::GridTools::PeriodicFacePair<dealii::TriaIterator<dealii::CellAccessor<2, 2> > > > >&’

How could I fix this ?

Reply:
You are right. The function takes a reference not a pointer. I guess you have to create an internal function `internal_create_triangulation`, which takes a pointer and is called by both `create_triangulation` functions.